remote_write.go

```diff
@@ -24,10 +27,38 @@ import (
 	"go.k6.io/k6/lib/netext/httpext"
 )
 
+var debugWriter *LockedWriter
+
 // Register the extension on module initialization, available to
 // import from JS as "k6/x/remotewrite".
 func init() {
 	modules.Register("k6/x/remotewrite", new(RemoteWrite))
+
+	const debugVar = "K6_PROMETHEUS_REMOTE_WRITE_DEBUG"
+	for _, envVar := range os.Environ() {
+		if len(envVar) > len(debugVar)+1 && envVar[:len(debugVar)+1] == debugVar+"=" {
+			debugFile := envVar[len(debugVar)+1:]
+			debugWriter = &LockedWriter{file: debugFile}
+			log.Printf("--- Debug output written to %s ---", debugFile)
+		}
+	}
+}
+
+type LockedWriter struct {
+	m    sync.Mutex
+	file string
+}
+
+func (lw *LockedWriter) Write(msg string) (n int, err error) {
+	lw.m.Lock()
+	defer lw.m.Unlock()
+
+	f, err := os.OpenFile(lw.file, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
I do feel like it will be better if we open it once on initializaiton

Reply:
I guess that makes it harder to figure out when to close it, but I doubt that is a problem, I am pretty sure that os.File doesn't have any buffering by default so everything will be in kernel buffers either way so :shrug: